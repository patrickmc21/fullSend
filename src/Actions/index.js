export const signInUser = (user) => ({
  type: 'SIGN_IN_USER',
  user
});

export const logoutUser = (id) => ({
  type: 'LOGOUT_USER',
  id
});

export const addRides = (rides) => ({
  type: 'ADD_RIDES',
  rides
});

export const updateRides = (rides) => ({
  type: 'UPDATE_RIDES',
  rides
});