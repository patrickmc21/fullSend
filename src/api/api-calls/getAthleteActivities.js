import moment from 'moment';
const startTime = moment().startOf('year');

export const getAthleteActivities = async (token, athleteId, start, end = Date.now()) => {
  const rootUrl = 'https://www.strava.com/api/v3/athlete/activities'
  const urlOptions = `?before=${end}&after=${start}&page=1`
  const optionsObject = {
    method: 'GET',
    headers: {
      Authorization: `Bearer ${token}`,
      "Content-Type": 'application/json' 
    }
  }
  const response = await fetch(`${rootUrl}${urlOptions}`, optionsObject);
  console.log(response);
  const data = await response.json();
  console.log(data);
  return data;
}