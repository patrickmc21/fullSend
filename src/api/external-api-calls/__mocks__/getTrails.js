import { mockActivityAndTrail } from '../../../mock-data/mock-data';

export const getTrails = jest.fn()
  .mockImplementation(() => {
    return Promise.resolve(mockActivityAndTrail)
  });