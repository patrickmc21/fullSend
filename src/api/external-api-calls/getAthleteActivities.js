export const getAthleteActivities = async (token, after, before) => {
  const rootUrl = 'https://www.strava.com/api/v3/athlete/activities'
  const urlOptions = `?before=${before}&after=${after}&page=1`
  const optionsObject = {
    method: 'GET',
    headers: {
      Authorization: `Bearer ${token}`,
      "Content-Type": 'application/json' 
    }
  };

  try {
    const response = await fetch(`${rootUrl}${urlOptions}`, optionsObject);
    const data = await response.json();
    return data;
  } catch (error) {
    throw error.message;
  }
}