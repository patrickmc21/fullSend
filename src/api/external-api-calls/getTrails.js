import key from '../apiKeys/mtb-project-key.js';

export const getTrails = async (lat, long) => {
  const rootUrl = 'https://www.mtbproject.com/data/get-trails?';
  const urlOptions = `lat=${lat}&lon=${long}&maxDistance=0&sort=distance&key=${key}`
  const response = await fetch(`${rootUrl}${urlOptions}`);
  const data = await response.json();
}