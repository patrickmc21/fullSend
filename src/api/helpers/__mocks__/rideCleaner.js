import { mockRides } from '../../../mock-data/mock-data';

const rideCleaner = jest.fn()
  .mockImplementation(() => {
    return mockRides
  });

  export default rideCleaner;