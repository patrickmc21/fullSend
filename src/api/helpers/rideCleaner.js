const rideCleaner = (ride) => {
  console.log(ride);
}