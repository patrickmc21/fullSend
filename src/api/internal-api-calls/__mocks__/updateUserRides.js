const updateUserRides = jest.fn();

export default updateUserRides;