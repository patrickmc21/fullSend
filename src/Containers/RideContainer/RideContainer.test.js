import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import { getAthleteActivities } from '../../api/external-api-calls/getAthleteActivities';
import { getTrails } from '../../api/external-api-calls/getTrails';
import updateUserRides from '../../api/internal-api-calls/updateUserRides';
import rideCleaner from '../../api/helpers/rideCleaner';
import * as actions from '../../Actions';
import { 
  RideContainer, 
  mapStateToProps, 
  mapDispatchToProps 
} from './RideContainer';
import * as mocks from '../../mock-data/mock-data';

jest.mock('../../api/external-api-calls/getAthleteActivities');
jest.mock('../../api/external-api-calls/getTrails');
jest.mock('../../api/internal-api-calls/updateUserRides');
jest.mock('../../api/helpers/rideCleaner');

describe('RideContainer', () => {

  let mockedUser;
  let mockedRides;
  let mockedUpdateRides;
  let wrapper;

  beforeEach(() => {
    mockedUser = mocks.mockUser;
    mockedRides = [];
    mockedUpdateRides = jest.fn();
    wrapper = shallow(
      <RideContainer
        user={mockedUser}
        rides={mockedRides}
        updateRides={mockedUpdateRides} />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});

describe('mapStateToProps', () => {

  let mapped;
  let mockedUser;
  let mockedRides;
  let mockedState;

  beforeEach(() => {
    mockedUser = mocks.mockUser;
    mockedRides = [mocks.mockRide];
    mockedState = {user: mockedUser, rides: mockedRides};
    mapped = mapStateToProps(mockedState);
  });


  it('should map user to props', () => {
    const expected = mockedUser;
    expect(mapped.user).toEqual(expected);
  });

  it('should map rides to props', () => {
    const expected = mockedRides;
    expect(mapped.rides).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {

  it('should map updateRides to props', () => {
    const mockedRides = [mocks.mockRide];
    const expected = {
      type: 'UPDATE_RIDES',
      rides: mockedRides
    };
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.updateRides(mockedRides);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });
});