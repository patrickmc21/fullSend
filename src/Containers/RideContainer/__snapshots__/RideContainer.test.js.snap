// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RideContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RideContainer
    month="April"
    rides={
        Array [
            Object {
              "date": "May 8th, 2017",
              "difficulty": "green",
              "distance": "2",
              "elapsedTime": "120",
              "epoch": "1523472606",
              "img": "someurlthing",
              "location": "park",
              "summary": "not sendy",
              "trailName": "lame",
              "userId": "9",
            },
            Object {
              "date": "May 14th, 2017",
              "difficulty": "blue",
              "distance": "5",
              "elapsedTime": "180",
              "epoch": "776232700",
              "img": "someurlthing",
              "location": "hill",
              "summary": "sendy",
              "trailName": "neat",
              "userId": "9",
            },
          ]
    }
    updateRides={[Function]}
    user={
        Object {
            "id": 12,
            "name": "Pat",
            "token": "2y589wyasdfhgfiouqehr8to8",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ride-container-nav"
>
          <withRouter(Connect(DateSelector)) />
          <button
                    className="update-rides"
                    onClick={[Function]}
          >
                    Update Rides
          </button>
</div>,
        <div
          className="card-container"
>
          <RideCard
                    ride={
                              Object {
                                        "date": "May 8th, 2017",
                                        "difficulty": "green",
                                        "distance": "2",
                                        "elapsedTime": "120",
                                        "epoch": "1523472606",
                                        "img": "someurlthing",
                                        "location": "park",
                                        "summary": "not sendy",
                                        "trailName": "lame",
                                        "userId": "9",
                                      }
                    }
          />
</div>,
      ],
      "className": "ride-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <withRouter(Connect(DateSelector)) />,
            <button
              className="update-rides"
              onClick={[Function]}
>
              Update Rides
</button>,
          ],
          "className": "ride-container-nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Update Rides",
              "className": "update-rides",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Update Rides",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [
              <RideCard
                ride={
                                Object {
                                                "date": "May 8th, 2017",
                                                "difficulty": "green",
                                                "distance": "2",
                                                "elapsedTime": "120",
                                                "epoch": "1523472606",
                                                "img": "someurlthing",
                                                "location": "park",
                                                "summary": "not sendy",
                                                "trailName": "lame",
                                                "userId": "9",
                                              }
                }
/>,
            ],
            false,
          ],
          "className": "card-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1523472606",
            "nodeType": "function",
            "props": Object {
              "ride": Object {
                "date": "May 8th, 2017",
                "difficulty": "green",
                "distance": "2",
                "elapsedTime": "120",
                "epoch": "1523472606",
                "img": "someurlthing",
                "location": "park",
                "summary": "not sendy",
                "trailName": "lame",
                "userId": "9",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ride-container-nav"
>
            <withRouter(Connect(DateSelector)) />
            <button
                        className="update-rides"
                        onClick={[Function]}
            >
                        Update Rides
            </button>
</div>,
          <div
            className="card-container"
>
            <RideCard
                        ride={
                                    Object {
                                                "date": "May 8th, 2017",
                                                "difficulty": "green",
                                                "distance": "2",
                                                "elapsedTime": "120",
                                                "epoch": "1523472606",
                                                "img": "someurlthing",
                                                "location": "park",
                                                "summary": "not sendy",
                                                "trailName": "lame",
                                                "userId": "9",
                                              }
                        }
            />
</div>,
        ],
        "className": "ride-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <withRouter(Connect(DateSelector)) />,
              <button
                className="update-rides"
                onClick={[Function]}
>
                Update Rides
</button>,
            ],
            "className": "ride-container-nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Update Rides",
                "className": "update-rides",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Update Rides",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [
                <RideCard
                  ride={
                                    Object {
                                                      "date": "May 8th, 2017",
                                                      "difficulty": "green",
                                                      "distance": "2",
                                                      "elapsedTime": "120",
                                                      "epoch": "1523472606",
                                                      "img": "someurlthing",
                                                      "location": "park",
                                                      "summary": "not sendy",
                                                      "trailName": "lame",
                                                      "userId": "9",
                                                    }
                  }
/>,
              ],
              false,
            ],
            "className": "card-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1523472606",
              "nodeType": "function",
              "props": Object {
                "ride": Object {
                  "date": "May 8th, 2017",
                  "difficulty": "green",
                  "distance": "2",
                  "elapsedTime": "120",
                  "epoch": "1523472606",
                  "img": "someurlthing",
                  "location": "park",
                  "summary": "not sendy",
                  "trailName": "lame",
                  "userId": "9",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`RideContainer should match the snapshot if no rides present 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RideContainer
    rides={Array []}
    updateRides={[Function]}
    user={
        Object {
            "id": 12,
            "name": "Pat",
            "token": "2y589wyasdfhgfiouqehr8to8",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ride-container-nav"
>
          <withRouter(Connect(DateSelector)) />
          <button
                    className="update-rides"
                    onClick={[Function]}
          >
                    Update Rides
          </button>
</div>,
        <div
          className="card-container"
>
          <h6
                    className="none-found"
          >
                    No Rides to Show!
          </h6>
</div>,
      ],
      "className": "ride-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <withRouter(Connect(DateSelector)) />,
            <button
              className="update-rides"
              onClick={[Function]}
>
              Update Rides
</button>,
          ],
          "className": "ride-container-nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Update Rides",
              "className": "update-rides",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Update Rides",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <h6
              className="none-found"
>
              No Rides to Show!
</h6>,
          ],
          "className": "card-container",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No Rides to Show!",
              "className": "none-found",
            },
            "ref": null,
            "rendered": "No Rides to Show!",
            "type": "h6",
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ride-container-nav"
>
            <withRouter(Connect(DateSelector)) />
            <button
                        className="update-rides"
                        onClick={[Function]}
            >
                        Update Rides
            </button>
</div>,
          <div
            className="card-container"
>
            <h6
                        className="none-found"
            >
                        No Rides to Show!
            </h6>
</div>,
        ],
        "className": "ride-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <withRouter(Connect(DateSelector)) />,
              <button
                className="update-rides"
                onClick={[Function]}
>
                Update Rides
</button>,
            ],
            "className": "ride-container-nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Update Rides",
                "className": "update-rides",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Update Rides",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <h6
                className="none-found"
>
                No Rides to Show!
</h6>,
            ],
            "className": "card-container",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "No Rides to Show!",
                "className": "none-found",
              },
              "ref": null,
              "rendered": "No Rides to Show!",
              "type": "h6",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`RideContainer should render all cards if month is All 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RideContainer
    month="All"
    rides={
        Array [
            Object {
              "date": "May 8th, 2017",
              "difficulty": "green",
              "distance": "2",
              "elapsedTime": "120",
              "epoch": "1523472606",
              "img": "someurlthing",
              "location": "park",
              "summary": "not sendy",
              "trailName": "lame",
              "userId": "9",
            },
            Object {
              "date": "May 14th, 2017",
              "difficulty": "blue",
              "distance": "5",
              "elapsedTime": "180",
              "epoch": "776232700",
              "img": "someurlthing",
              "location": "hill",
              "summary": "sendy",
              "trailName": "neat",
              "userId": "9",
            },
          ]
    }
    updateRides={[Function]}
    user={
        Object {
            "id": 12,
            "name": "Pat",
            "token": "2y589wyasdfhgfiouqehr8to8",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ride-container-nav"
>
          <withRouter(Connect(DateSelector)) />
          <button
                    className="update-rides"
                    onClick={[Function]}
          >
                    Update Rides
          </button>
</div>,
        <div
          className="card-container"
>
          <RideCard
                    ride={
                              Object {
                                        "date": "May 8th, 2017",
                                        "difficulty": "green",
                                        "distance": "2",
                                        "elapsedTime": "120",
                                        "epoch": "1523472606",
                                        "img": "someurlthing",
                                        "location": "park",
                                        "summary": "not sendy",
                                        "trailName": "lame",
                                        "userId": "9",
                                      }
                    }
          />
          <RideCard
                    ride={
                              Object {
                                        "date": "May 14th, 2017",
                                        "difficulty": "blue",
                                        "distance": "5",
                                        "elapsedTime": "180",
                                        "epoch": "776232700",
                                        "img": "someurlthing",
                                        "location": "hill",
                                        "summary": "sendy",
                                        "trailName": "neat",
                                        "userId": "9",
                                      }
                    }
          />
</div>,
      ],
      "className": "ride-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <withRouter(Connect(DateSelector)) />,
            <button
              className="update-rides"
              onClick={[Function]}
>
              Update Rides
</button>,
          ],
          "className": "ride-container-nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Update Rides",
              "className": "update-rides",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Update Rides",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [
              <RideCard
                ride={
                                Object {
                                                "date": "May 8th, 2017",
                                                "difficulty": "green",
                                                "distance": "2",
                                                "elapsedTime": "120",
                                                "epoch": "1523472606",
                                                "img": "someurlthing",
                                                "location": "park",
                                                "summary": "not sendy",
                                                "trailName": "lame",
                                                "userId": "9",
                                              }
                }
/>,
              <RideCard
                ride={
                                Object {
                                                "date": "May 14th, 2017",
                                                "difficulty": "blue",
                                                "distance": "5",
                                                "elapsedTime": "180",
                                                "epoch": "776232700",
                                                "img": "someurlthing",
                                                "location": "hill",
                                                "summary": "sendy",
                                                "trailName": "neat",
                                                "userId": "9",
                                              }
                }
/>,
            ],
            false,
          ],
          "className": "card-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1523472606",
            "nodeType": "function",
            "props": Object {
              "ride": Object {
                "date": "May 8th, 2017",
                "difficulty": "green",
                "distance": "2",
                "elapsedTime": "120",
                "epoch": "1523472606",
                "img": "someurlthing",
                "location": "park",
                "summary": "not sendy",
                "trailName": "lame",
                "userId": "9",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "776232700",
            "nodeType": "function",
            "props": Object {
              "ride": Object {
                "date": "May 14th, 2017",
                "difficulty": "blue",
                "distance": "5",
                "elapsedTime": "180",
                "epoch": "776232700",
                "img": "someurlthing",
                "location": "hill",
                "summary": "sendy",
                "trailName": "neat",
                "userId": "9",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ride-container-nav"
>
            <withRouter(Connect(DateSelector)) />
            <button
                        className="update-rides"
                        onClick={[Function]}
            >
                        Update Rides
            </button>
</div>,
          <div
            className="card-container"
>
            <RideCard
                        ride={
                                    Object {
                                                "date": "May 8th, 2017",
                                                "difficulty": "green",
                                                "distance": "2",
                                                "elapsedTime": "120",
                                                "epoch": "1523472606",
                                                "img": "someurlthing",
                                                "location": "park",
                                                "summary": "not sendy",
                                                "trailName": "lame",
                                                "userId": "9",
                                              }
                        }
            />
            <RideCard
                        ride={
                                    Object {
                                                "date": "May 14th, 2017",
                                                "difficulty": "blue",
                                                "distance": "5",
                                                "elapsedTime": "180",
                                                "epoch": "776232700",
                                                "img": "someurlthing",
                                                "location": "hill",
                                                "summary": "sendy",
                                                "trailName": "neat",
                                                "userId": "9",
                                              }
                        }
            />
</div>,
        ],
        "className": "ride-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <withRouter(Connect(DateSelector)) />,
              <button
                className="update-rides"
                onClick={[Function]}
>
                Update Rides
</button>,
            ],
            "className": "ride-container-nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Update Rides",
                "className": "update-rides",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Update Rides",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [
                <RideCard
                  ride={
                                    Object {
                                                      "date": "May 8th, 2017",
                                                      "difficulty": "green",
                                                      "distance": "2",
                                                      "elapsedTime": "120",
                                                      "epoch": "1523472606",
                                                      "img": "someurlthing",
                                                      "location": "park",
                                                      "summary": "not sendy",
                                                      "trailName": "lame",
                                                      "userId": "9",
                                                    }
                  }
/>,
                <RideCard
                  ride={
                                    Object {
                                                      "date": "May 14th, 2017",
                                                      "difficulty": "blue",
                                                      "distance": "5",
                                                      "elapsedTime": "180",
                                                      "epoch": "776232700",
                                                      "img": "someurlthing",
                                                      "location": "hill",
                                                      "summary": "sendy",
                                                      "trailName": "neat",
                                                      "userId": "9",
                                                    }
                  }
/>,
              ],
              false,
            ],
            "className": "card-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1523472606",
              "nodeType": "function",
              "props": Object {
                "ride": Object {
                  "date": "May 8th, 2017",
                  "difficulty": "green",
                  "distance": "2",
                  "elapsedTime": "120",
                  "epoch": "1523472606",
                  "img": "someurlthing",
                  "location": "park",
                  "summary": "not sendy",
                  "trailName": "lame",
                  "userId": "9",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "776232700",
              "nodeType": "function",
              "props": Object {
                "ride": Object {
                  "date": "May 14th, 2017",
                  "difficulty": "blue",
                  "distance": "5",
                  "elapsedTime": "180",
                  "epoch": "776232700",
                  "img": "someurlthing",
                  "location": "hill",
                  "summary": "sendy",
                  "trailName": "neat",
                  "userId": "9",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
